pynicotine/external/update_ipcountry_data.py

```diff
@@ -0,0 +1,95 @@
+# COPYRIGHT (C) 2024 Nicotine+ Contributors
+#
+# GNU GENERAL PUBLIC LICENSE
+#    Version 3, 29 June 2007
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import csv
+import io
+import os
+import urllib.request
+import zipfile
+
+
+COUNTRY_DB_URL = "https://download.ip2location.com/lite/IP2LOCATION-LITE-DB1.CSV.ZIP"
+FILE_PATH, _HEADERS = urllib.request.urlretrieve(COUNTRY_DB_URL)
+CURRENT_PATH = os.path.dirname(os.path.realpath(__file__))
+MAX_IPV4_RANGE = 4294967295
+
+
+def update_ip_country_data():
+    """Dowmload IP2Location country data and convert it for use in Python."""
+
+    output = bytearray(b"""# IP2Location LITE (c) 2001-2024 Hexasoft Development Sdn. Bhd.
+
+    # IP2Location LITE is licensed under a
+    # Creative Commons Attribution-ShareAlike 4.0 International License.
+
+    # You should have received a copy of the license along with this
+    # work. If not, see <http://creativecommons.org/licenses/by-sa/4.0/>.
+
+    """)
+    country_codes = set()
+    ip_ranges = []
+    country_values = []
+
+    with zipfile.ZipFile(FILE_PATH, "r") as zip_file_handle:
+        with zip_file_handle.open(zip_file_handle.namelist()[-1]) as csv_file_handle:
+            for row in csv.reader(io.TextIOWrapper(csv_file_handle)):
```

Comment:
Perhaps the values could be loaded directly from a local copy of the csv file at runtime?